thingif/src/main/java/com/kii/thingif/ThingIFAPI.java

```diff
@@ -1088,19 +1088,17 @@ private Trigger postServerCodeNewTrigger(
         if (predicate == null) {
             throw new IllegalArgumentException("predicate is null");
         }
-        //TODO: // FIXME: 2017/01/23
-        return null;
-//        JSONObject requestBody = options != null ?
-//                JsonUtils.newJson(GsonRepository.gson().toJson(options)) :
-//                new JSONObject();
-//        try {
-//            requestBody.put("predicate", JsonUtils.newJson(GsonRepository.gson().toJson(predicate)));
-//            requestBody.put("triggersWhat", TriggersWhat.SERVER_CODE.name());
-//            requestBody.put("serverCode", JsonUtils.newJson(GsonRepository.gson().toJson(serverCode)));
-//        } catch (JSONException e) {
-//            // Won't happen
-//        }
-//        return this.postNewTrigger(requestBody);
+        JSONObject requestBody = options != null ?
+                JsonUtils.newJson(this.gson.toJson(options)) :
+                new JSONObject();
+        try {
+            requestBody.put("predicate", JsonUtils.newJson(this.gson.toJson(predicate, Predicate.class)));
+            requestBody.put("triggersWhat", TriggersWhat.SERVER_CODE.name());
+            requestBody.put("serverCode", JsonUtils.newJson(this.gson.toJson(serverCode)));
+        } catch (JSONException e) {
+            // Won't happen
+        }
```

Comment:
Like [here](https://github.com/KiiPlatform/thing-if-AndroidSDK/pull/235/files#diff-12e2b8b1411bb1a56fce72fd0a2187a8R1023), I prefer to throw `RuntimeException` even if that situation does not occur.